A compiler warning in preamble.lua.c: string literal of length 6738  exceeds maximum length 4095 that ISO C99 compilers are required to support

`ninja luaL_loadbuffer_proto_test`:

```
<snipped>

[8/9] Building C object tests/capi/luaL_loadbuffer_proto/CMakeFiles/luaL_loadbuffer_proto_test.dir/preamble.lua.c.o
/home/sergeyb/sources/lunapark/build/lua/tests/capi/luaL_loadbuffer_proto/preamble.lua.c:2:1: warning: string literal of length 6738 exceeds maximum length 4095 that ISO C99 compilers are required to support [-Woverlength-strings]
    2 | "local DEFAULT_NUMBER = 1\n"
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
    3 | "local GLOBALS = {}\n"
      | ~~~~~~~~~~~~~~~~~~~~~~
    4 | "\n"
      | ~~~~
    5 | "local always_number = function(val)\n"
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    6 | "    return tonumber(val) or DEFAULT_NUMBER\n"
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    7 | "end\n"
      | ~~~~~~~
    8 | "\n"
      | ~~~~
    9 | "---@diagnostic disable-next-line: unused\n"
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   10 | "local not_nan_and_nil = function(val)\n"
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   11 | "    return (val ~= val or val == nil) and DEFAULT_NUMBER or val\n"
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   12 | "end\n"
      | ~~~~~~~
   13 | "\n"
      | ~~~~
   14 | "_G.always_number = always_number\n"
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   15 | "\n"
      | ~~~~
   16 | "local __add = function(v1, v2)\n"
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   17 | "    return always_number(v1) + always_number(v2)\n"
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
[9/9] Linking CXX executable tests/capi/luaL_loadbuffer_proto/luaL_loadbuffer_proto_test
```

--- a/tests/capi/luaL_loadbuffer_proto/CMakeLists.txt
+++ b/tests/capi/luaL_loadbuffer_proto/CMakeLists.txt
@@ -5,6 +5,8 @@ endif ()
 
 set(test_lua_sources)
 lua_source(test_lua_sources preamble.lua preamble_lua)
+set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/preamble.lua.c
+    PROPERTIES COMPILE_OPTIONS "-Wno-overlength-strings")
 
 add_custom_target(generate_test_lua_sources
     WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/tests/capi/luaL_loadbuffer_proto/

